Mathlib/Analysis/ODE/PicardLindelof.lean

```diff
@@ -10,397 +10,590 @@ import Mathlib.Topology.MetricSpace.Contracting
-theorem exists_forall_hasDerivAt_Ioo_eq_of_contDiffAt (hv : ContDiffAt ℝ 1 v x₀) :
-    ∃ f : ℝ → E, f t₀ = x₀ ∧
-      ∃ ε > (0 : ℝ), ∀ t ∈ Ioo (t₀ - ε) (t₀ + ε), HasDerivAt f (v (f t)) t := by
-  obtain ⟨ε, hε, _, _, _, hpl⟩ := exists_isPicardLindelof_const_of_contDiffAt t₀ hv
-  obtain ⟨f, hf1, hf2⟩ := hpl.exists_forall_hasDerivWithinAt_Icc_eq x₀
-  exact ⟨f, hf1, ε, hε, fun t ht =>
-    (hf2 t (Ioo_subset_Icc_self ht)).hasDerivAt (Icc_mem_nhds ht.1 ht.2)⟩
-
-/-- A time-independent, continuously differentiable ODE admits a solution in some open interval. -/
-theorem exists_forall_hasDerivAt_Ioo_eq_of_contDiff (hv : ContDiff ℝ 1 v) :
-    ∃ f : ℝ → E, f t₀ = x₀ ∧
-      ∃ ε > (0 : ℝ), ∀ t ∈ Ioo (t₀ - ε) (t₀ + ε), HasDerivAt f (v (f t)) t :=
-  let ⟨f, hf1, ε, hε, hf2⟩ :=
-    exists_forall_hasDerivAt_Ioo_eq_of_contDiffAt t₀ hv.contDiffAt
-  ⟨f, hf1, ε, hε, fun _ h => hf2 _ h⟩
+Picard-Lindelöf theorem. -/
+lemma of_contDiffAt_one [NormedSpace ℝ E]
+    {f : E → E} {x₀ : E} (hf : ContDiffAt ℝ 1 f x₀) (t₀ : ℝ) :
+    ∃ (ε : ℝ) (hε : 0 < ε) (a r L K : ℝ≥0) (_ : 0 < r), IsPicardLindelof (fun _ ↦ f)
+      (tmin := t₀ - ε) (tmax := t₀ + ε) ⟨t₀, (by simp [le_of_lt hε])⟩ x₀ a r L K := by
+  -- Obtain ball of radius `a` within the domain in which f is `K`-lipschitz
+  obtain ⟨K, s, hs, hl⟩ := hf.exists_lipschitzOnWith
+  obtain ⟨a, ha : 0 < a, has⟩ := Metric.mem_nhds_iff.mp hs
+  set L := K * a + ‖f x₀‖ + 1 with hL
+  have hL0 : 0 < L := by positivity
+  have hb (x : E) (hx : x ∈ closedBall x₀ (a / 2)) : ‖f x‖ ≤ L := by
+    rw [hL]
+    calc
+      ‖f x‖ ≤ ‖f x - f x₀‖ + ‖f x₀‖ := norm_le_norm_sub_add _ _
+      _ ≤ K * ‖x - x₀‖ + ‖f x₀‖ := by
+        gcongr
+        rw [← dist_eq_norm, ← dist_eq_norm]
+        apply hl.dist_le_mul x _ x₀ (mem_of_mem_nhds hs)
+        apply subset_trans _ has hx
+        exact closedBall_subset_ball <| half_lt_self ha -- this is where we need `a / 2`
+      _ ≤ K * a + ‖f x₀‖ := by
+        gcongr
+        rw [← mem_closedBall_iff_norm]
+        exact closedBall_subset_closedBall (half_le_self (le_of_lt ha)) hx
+      _ ≤ L := le_add_of_nonneg_right zero_le_one
+  let ε := a / L / 2 / 2
+  have hε0 : 0 < ε := by positivity
+  refine ⟨ε, hε0,
+    ⟨a / 2, le_of_lt <| half_pos ha⟩, ⟨a / 2, le_of_lt <| half_pos ha⟩ / 2,
+    ⟨L, le_of_lt hL0⟩, K, half_pos <| half_pos ha, ?_⟩
+  apply of_time_independent hb <|
+    hl.mono <| subset_trans (closedBall_subset_ball (half_lt_self ha)) has
+  rw [NNReal.coe_mk, add_sub_cancel_left, sub_sub_cancel, max_self, NNReal.coe_div,
+    NNReal.coe_two, NNReal.coe_mk, mul_comm, ← le_div_iff₀ hL0, sub_half, div_right_comm (a / 2),
+    div_right_comm a]
+
+end
+
+/-! ## Existence of solutions to ODEs -/
+
+open ODE
+
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] [CompleteSpace E]
+  {f : ℝ → E → E} {tmin tmax : ℝ} {t₀ : Icc tmin tmax} {x₀ x : E} {a r L K : ℝ≥0}
+
+/-- Picard-Lindelöf (Cauchy-Lipschitz) theorem, integral form. This version shows the existence of a
```

Comment:
One final request: can you bolden the theorem name (e.g., `**Picard-Lindelöf (Cauchy-Lipschitz) theorem**`; I don't mind if the "integral form" is bold or not). If I understand correctly, this is what mathlib style tries to gravitate to.

I appreciate this is pre-existing - so saying "yes, but in a follow-up PR" is fine.